bin/dump_data_to_file.py

```diff
@@ -0,0 +1,115 @@
+import json
+import os
+import requests
+import time
+import arrow
+import argparse
+import sys; sys.path.append("..")
+from emeval.input.spec_details import ServerSpecDetails
+from emeval.input.phone_view import PhoneView
+
+
+DEFAULT_SPEC_USER = "shankari@eecs.berkeley.edu"
+DATASTORE_URL = None
+
+def dump_to_file(data, spec_id, user, key, start_ts, end_ts):
+    """
+    Dumped outputs are created recursively in folder relative to path of script.
+    """
+    key = key.replace("/", "~") # key could have a slash, replace with tilde
+
+    out_path = os.path.join("data", spec_id, user, key)
+    os.makedirs(out_path, exist_ok=True)
+
+    out_file = os.path.join(out_path, f"{start_ts}_{end_ts}.json")
+
+    print(f"Creating {out_file=}...")
+
+    with open(out_file, "w") as f:
+        json.dump(data, f, indent=4)
+
+
+def retrieve_data_from_server(user, key, start_ts, end_ts):
```

Comment:
I really don't like copy-pasting code. DRY (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). This is almost identical to `retrieve_data_from_server` in `ServerSpecDetails`, why not use that directly? The use cases might be slightly different and need some minor refactoring, but you should not have to copy-paste the entire function.

Reply:
Note that this doesn't even work properly because it doesn't handle the batching that is required for certain keys (i.e. `read_until_done`). See more detailed comment below.